EIPS/eip-7378.md

```diff
@@ -0,0 +1,75 @@
+
+A [theorethical work](../assets/eip-7378/AGHH.pdf) describes how both miners and users can initiate an attack. For example, we can imagine that users who wish to pay lower costs will coordinate the attack. Roughly, a user (or group of users) that has transactions with a total $g$ amount of gas bribes the miner of the current block (no matter the miner's power) to propose an empty block instead. The cost of such a bribe is only $\varepsilon \times {s^* }$ -- the tip times the target block size. Consequently, the base fee reduces in the next block. If we accept that EIP-1559 reaches its goals, e.g., users would typicaly use a simple and honest bidding strategy of reporting their maximal willingness to pay plus adding a small tip ($\varepsilon$), then in the honest users' steady state, gas proposals leave the miners with an $\varepsilon$ tip. Given that other users are na\"ive (or slow to react), our bribing user will include its transactions with any tip larger than $\varepsilon$ -- making the attack profitable whenever $g \frac{b^* }{8} >s^* \varepsilon$.
+
+## Specification
+
+The following equation describes the main part of the change, that is, replacing $s[i]$ by $s_{\textit{avg}}[i]$. For $q\in (0,1)$,
+
+$$s_{\textit{avg}}[i] \triangleq \frac{1-q}{q}\sum_{k=1}^{\infty } q^k\cdot s[i-k+1]$$ 
+
+which simplifies to the recursive form
+$$s_{\textit{avg}}[i] = (1-q)\cdot s[i] + q\cdot s_{\textit{avg} }[i-1].$$
+
+This scheme is parametrized by $q$ which determins the concentration of the average. Roughly, smaller $q$ values result in $s_\textit{avg}$ resembling the very last block sizes, while larger $q$ values smoothen the average over a longer history.
+
+
+## Rationale
+
+An intuitive option for the Transaction Fee Mechanism (TFM) that adjusts supply and demand economically is *First price auction*, which is well known and studied. Nevertheless, the Ethereum network choice was to use EIP-1559 for the TFM (one stated reason was to try and simplify the fee estimation for users, and reduce the advantage of sophisticated users). In this proposal, our design goal is to improve the TFM (of EIP-1559) by mitigating known problems that it raises. It is important to note that these problems severity are in direct relation to the demand for block space, and currently only mildly impact the Ethereum network. If demand to use Ethereum increases, however, these problems are expected to exacerbate. We may want to prepare for this beforehand.
+
+The change is based on [this work](../assets/eip-7378/AGHH.pdf) that described a rational strategy in which bribes are profitabe. Choosing to average based on a geometric series weights results in two desired properties: (i) the computation and space complexity are both in O(1), and (ii) the average gradually phases out the impact of a single outlier block without causing significant future fluctuations in the base fee.
```

Comment:
Is there an analysis of smoothening that is proposed by this eip vs smoothening that can be done by base fee max change denominator?

Reply:
I apologize if I misinterpret the comment regarding the max change denominator. If the meaning is to reduce the learning rate $\phi$ from 1/8 to something smaller, then this is problematic as it significantly affects the responsiveness of the base fee, making it slow to respond to actual persisting changes. The reason for using geometric series weights is precisely to obtain the favorable tradeoff that still responds quickly while mitigating the incentive misalignments.

Reply:
yes, i don't disagree that geometric weights are better, a short analysis /sim run like how it has been done for other things in the pdf would sort of complete the proposal that it can't be achieved with making the `max change denominator` changes. in any case, `max change denominator` argument needs to be included in the rational (like how you have articulated above)